File: Data-Structures-And-Libraries/Python/URI-1025.py
def find(dictionary: "dict of stones", value: "stone to find") -> "identifier":
    if value in dictionary.keys():
        return "{} found at {}".format(value, dictionary[value])

    else:
        return "{} not found".format(value)


def main():

    global_case = 1

    while True:
        entry = input().split()
        q, n = int(entry[0]), int(entry[1])
        if q + n == 0:
            break
        stones = [None] * q
        for i in range(len(stones)):
            stones[i] = int(input())
        stones.sort()

        stones_dict = {}
        counter = 1
        for stone in stones:
            if stone not in stones_dict.keys():
                stones_dict[stone] = counter
            counter += 1

        print("CASE# {}:".format(global_case))
        for i in range(n):
            value = int(input())
            ans = find(stones_dict, value)
            print(ans)

        global_case += 1


main()

File: Data-Structures-And-Libraries/Java/URI-1025.java
import java.io.IOException;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Scanner;

public class Main {
 
    public static void main(String[] args) throws IOException {
        
        Scanner sc2 = new Scanner(System.in);
		int global = 1;
 
		while(true) {
 
			int q = sc2.nextInt();
			int n = sc2.nextInt();
 
			if(q + n == 0) {
				break;
 
			}
 
			sc2.nextLine();
 
			int[] stones = new int[q];
 
			for(int i = 0; i < stones.length; i++) {
				stones[i] = sc2.nextInt();
				sc2.nextLine();
 
			}
 
			Arrays.sort(stones);
			HashMap<Integer, Integer> dictionary= new HashMap<Integer, Integer>();
 
			for (int i = 0; i < stones.length; i++) { 			
				int key = stones[i];
				if(dictionary.get(key) == null) { 
					dictionary.put(key, i + 1); 
				}						  
			 }	
			/* System.out.println(dictionary.entrySet()); */
			System.out.printf("CASE# %d:%n", global);
 
			for (int i = 0; i < n; i++) {
 
				int value = sc2.nextInt(); 
				String ans = find(dictionary,value);
				System.out.println(ans);
				sc2.nextLine();
 
			  }
 
			global += 1;
 
		}		
		sc2.close();
	}
 
	static String find(HashMap<Integer, Integer> dictionary, int value) {
		if(dictionary.get(value) != null) {
			return String.format("%d found at %d", value, dictionary.get(value));
 
		}
 
		else {
			return String.format("%d not found", value);
 
		}
        
    }
 
}
